sources/standard/filesystem/__init__.py

```diff
@@ -0,0 +1,178 @@
+"""This source collects fsspec files and downloads or proccess them."""
+import mimetypes
+import os
+from typing import Optional
+
+import dlt
+from dlt.common import pendulum
+from dlt.common.storages import filesystem
+from dlt.common.storages.configuration import FilesystemConfiguration
+from dlt.common.time import ensure_pendulum_datetime
+from dlt.extract.source import DltResource, TDataItem, TDataItems
+
+from ..file_source import FileModel
+from .settings import DEFAULT_CHUNK_SIZE, DEFAULT_START_DATE, STORAGE_PATH
+
+from fsspec import AbstractFileSystem
+import pandas as pd
+
+
+@dlt.source
+def filesystem_source(
+    credentials: FilesystemConfiguration = dlt.secrets.value,
+    storage_path: str = STORAGE_PATH,
+    start_date: pendulum.DateTime = DEFAULT_START_DATE,
+    filename_filter: Optional[str] = None,
+    chunksize: int = DEFAULT_CHUNK_SIZE,
+    extract_content: bool = False,
```

Comment:
let's make it `load_content` and if set, they content of the file will be loaded into memory (`file_content` of FileModel) and yielded. we should load as bytes. opening string reader over bytes is trivial